hearpreprocess/pipeline.py

```diff
@@ -530,6 +522,69 @@ def split_train_test_val(self, metadata: pd.DataFrame):
         metadata.loc[metadata["split_key"].isin(test_split_keys), "split"] = "test"
         return metadata
 
+    def assert_correct_kfolds(self, metadata: pd.DataFrame):
+        """
+        Raises an AssertionError if the set of fold names in task_config doesn't
+        match the set of fold names used for splits in the metadata
+        """
+        if set(metadata["split"].unique()) != set(self.task_config["splits"]):
+            raise AssertionError(
+                "Names of splits in metadata don't match the required names for a "
+                f"kfold dataset. Expected: {self.task_config['splits']}. "
+                f"Received: {metadata['split'].unique()}."
+            )
+
+    def split_k_folds(self, metadata: pd.DataFrame):
+        """
+        Deterministically split dataset into k-folds
+        """
+
+        splits_present = metadata["split"].unique()
+        if len(splits_present) > 1:
```

Comment:
Again this logic is really confusing. We need some documentation *in one place* of what the expected behavior is for kfold config.py etc. It's really too hard to follow this based upon the code.